### DISCUSSION TITLE
Create little script

### DISCUSSION THREAD

**Discussion by Mara-Li** [SilentVoid13/Templater#219] [Help]
Hello! I start to understand using templater (yes, after so much time...)

How can I save little script to just use like a function in my YAML, or my notes, somethings like `tp.user.my_function(value)`? I have some question about that :
- Does it need to be a `.js` file ? 
- I read that the user script can't read/use `tp`, and `tR`. So, need I recreate the function in?
- Have you some examples of an existing script? 

If you want to help me step by step, my @ on Discord is @Mara#3000, and I will be very pleased to have little help!

**Comment by AB1908** [comment]
Is this still an issue? You can look at some of shabegom's snippets: https://shbgm.ca/obsidian/docs/how-to-use-templater-js-scripts.

It isn't quite necessary to use user scripts though. It's just a neat way of storing things.
